staging/src/k8s.io/api/autoscaling/v2/types.go

```diff
@@ -361,7 +360,7 @@ type HorizontalPodAutoscalerStatus struct {
 
 	// currentReplicas is current number of replicas of pods managed by this autoscaler,
 	// as last seen by the autoscaler.
-	CurrentReplicas int32 `json:"currentReplicas" protobuf:"varint,3,opt,name=currentReplicas"`
+	CurrentReplicas *int32 `json:"currentReplicas" protobuf:"varint,3,opt,name=currentReplicas"`
 
 	// desiredReplicas is the desired number of replicas of pods managed by this autoscaler,
 	// as last calculated by the autoscaler.
```

Comment:
note that `conditions` is being made `+optional` in v2beta{1,2} in https://github.com/kubernetes/kubernetes/pull/100318 to fix a bug with client-side validation

it should at least be made `+optional` here, and probably set to omitempty as well

Reply:
Ok, THKs.